### PR TITLE
fix: make sure AVS calls use qualified/non-qualified identifiers based on federation FS-506

### PR DESCRIPTION
**Pull request by marcoconti83**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-506" title="FS-506" target="_blank">BugFS-506</a>  [Android] Issues with API versioning staging
  </td></table>
  <br />
 

#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

With https://github.com/wireapp/wire-android/pull/3695 I fixed the fact that calling was not working on federation-enabled environments. However, it was still not working on non-federation environments when calling from Android to any other platform.

### Causes

The Android code was always using fully qualified identifiers even if the environment is non federated. Other platforms correctly do not use fully qualified identifiers in these environments, therefore Android was not able to establish call with other platforms.

### Solutions

There is a lot of code related to calling. I hope I captured all the points where we define qualified identifiers (conversation and clients) and added a check for federation support based on up-to-date federation support status.

### Testing

I manually tested by calling in 1:1 from Android to Web on both staging (non-federated) and Anta (federated). It worked.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build Success
ID:1
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3698/1/)

File: zmessaging/src/main/scala/com/waz/model/RConvQualifiedId.scala [hunk 1]
```diff
@@ -13,6 +13,7 @@ final case class RConvQualifiedId(id: RConvId, domain: String) {
 object RConvQualifiedId {
   def apply(id: RConvId): RConvQualifiedId = RConvQualifiedId(id, "")
   def apply(id: RConvId, domain: Domain): RConvQualifiedId = RConvQualifiedId(id, domain.str)
+  def apply(id: RConvId, domain: Domain, useDomain: Boolean): RConvQualifiedId = if(useDomain) apply(id) else apply(id, domain)
```
[the remark this answers is not in the document]
Comment (marcoconti83) [reply zmessaging/src/main/scala/com/waz/model/RConvQualifiedId.scala]: Fixed

**Comment by AndroidBob** [comment]
Build Success
ID:2
URL:[Link to Buildjob](https://10.10.124.134/job/wire-android/job/PR-3698/2/)

**Review by vitorhugods — APPROVED** [review]
✅

**Review by ohassine — APPROVED** [review]

**Pull request merged**